test/e2e/reconciler/ingress/controller_test.go

```diff
@@ -0,0 +1,594 @@
+
+				ingressYaml, err := rawCustomResourceDefinitions.ReadFile("ingress.yaml")
+				if err != nil {
+					t.Errorf("failed to read ingress: %v", err)
+					return
+				}
+
+				var ingress *v1.Ingress
+				if err := yaml.Unmarshal(ingressYaml, &ingress); err != nil {
+					t.Errorf("failed to create ingress: %v", err)
+					return
+				}
+
+				ingress, err = servers[sourceClusterName].client.Create(ctx, ingress, metav1.CreateOptions{})
+				if err != nil {
+					t.Errorf("failed to create ingress: %v", err)
+					return
+				}
+
+				ingress.Name = ingress.Name + "--" + clusterName
+				if err := servers[sinkClusterName].expect(ingress, func(object *v1.Ingress) error {
+					if diff := cmp.Diff(ingress.Spec, object.Spec); diff != "" {
+						return fmt.Errorf("saw incorrect spec on sink cluster: %s", diff)
+					}
+					return nil
+				}); err != nil {
+					t.Errorf("did not see Ingress spec updated on sink cluster: %v", err)
+					return
+				}
+			},
+		},
+		{
+			name: "update the ingress expect the sink to be updated",
+			work: func(ctx context.Context, t framework.TestingTInterface, servers map[string]runningServer) {
+				ingressYaml, err := rawCustomResourceDefinitions.ReadFile("ingress.yaml")
+				if err != nil {
+					t.Errorf("failed to read ingress: %v", err)
+					return
+				}
+
+				var ingress *v1.Ingress
+				if err := yaml.Unmarshal(ingressYaml, &ingress); err != nil {
+					t.Errorf("failed to create ingress: %v", err)
+					return
+				}
+
+				ingress, err = servers[sourceClusterName].client.Create(ctx, ingress, metav1.CreateOptions{})
+				if err != nil {
+					t.Errorf("failed to create ingress: %v", err)
+					return
+				}
+
+				expectedIngress := ingress.DeepCopy()
+				expectedIngress.Name = expectedIngress.Name + "--" + clusterName
+				if err := servers[sinkClusterName].expect(expectedIngress, func(object *v1.Ingress) error {
+					if diff := cmp.Diff(expectedIngress.Spec, object.Spec); diff != "" {
+						return fmt.Errorf("saw incorrect spec on sink cluster: %s", diff)
+					}
+					return nil
+				}); err != nil {
+					t.Errorf("did not see the ingress synced on sink cluster: %v", err)
+					return
+				}
+
+				err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					ingress, err := servers[sourceClusterName].client.Get(ctx, ingress.Name, metav1.GetOptions{})
+					if err != nil {
+						return err
+					}
+					ingress.Spec.Rules[0].Host = "valid-ingress-2.kcp-apps.127.0.0.1.nip.io"
+					_, err = servers[sourceClusterName].client.Update(ctx, ingress, metav1.UpdateOptions{})
+					if err != nil {
+						return err
+					}
+					return nil
+				})
+
+				if err != nil {
+					t.Errorf("failed updating the ingress object in the source cluster: %v", err)
+					return
+				}
+
+				ingress.Name = ingress.Name + "--" + clusterName
+				if err := servers[sinkClusterName].expect(ingress, func(object *v1.Ingress) error {
+					if diff := cmp.Diff(ingress.Spec, object.Spec); diff != "" {
```

Comment:
nit: you are only looking for the updated host, right? would likely make sense to somehow scope down the check for that only so it's a bit clearer what the test intent is for whoever breaks the test :)